handel/src/services/stepfunctions/index.ts

```diff
@@ -0,0 +1,138 @@
+    } else if (path.extname(serviceContext.params.definition) === '.json') {
+        if (util.readJsonFileSync(serviceContext.params.definition) === null) {
+            errors.push(`${SERVICE_NAME} - ${serviceContext.params.definition} is not a valid JSON file.`);
+        }
+    } else if (['.yml', '.yaml'].includes(path.extname(serviceContext.params.definition))) {
+        if (util.readYamlFileSync(serviceContext.params.definition) === null) {
+            errors.push(`${SERVICE_NAME} - ${serviceContext.params.definition} is not a valid YAML file.`);
+        }
+    } else {
+        errors.push(`${SERVICE_NAME} - The 'definition' parameter must have file extension .json, .yml, or .yaml.`);
+    }
+
+    return errors;
+}
+
+export async function deploy(ownServiceContext: ServiceContext<StepFunctionsConfig>, ownPreDeployContext: PreDeployContext, dependenciesDeployContexts: DeployContext[]): Promise<DeployContext> {
+    const stackName = deployPhaseCommon.getResourceName(ownServiceContext);
+    winston.info(`${SERVICE_NAME} - Executing Deploy on '${stackName}'`);
+    const compiledStepFunctionsTemplate = await getCompiledStepFunctionsTemplate(stackName, ownServiceContext, dependenciesDeployContexts);
+    const stackTags = getTags(ownServiceContext);
+    const deployedStack = await deployPhaseCommon.deployCloudFormationStack(stackName, compiledStepFunctionsTemplate, [], true, SERVICE_NAME, 30, stackTags);
+    winston.info(`${SERVICE_NAME} - Finished deploying '${stackName}'`);
+    return getDeployContext(ownServiceContext, deployedStack);
+}
+
+export async function unDeploy(ownServiceContext: ServiceContext<StepFunctionsConfig>): Promise<UnDeployContext> {
+    await iamCalls.detachPoliciesFromRole(deployPhaseCommon.getResourceName(ownServiceContext));
+    return deletePhasesCommon.unDeployService(ownServiceContext, SERVICE_NAME);
+}
+
+export const producedEventsSupportedServices = [];
+
+export const producedDeployOutputTypes = ['environmentVariables', 'policies'];
+
+export const consumedDeployOutputTypes = ['environmentVariables', 'policies'];
+
+export const supportsTagging = false;
+
+function generateDefinitionString(filename: string, dependenciesDeployContexts: DeployContext[]): string {
+    const readFile = path.extname(filename) == '.json' ? util.readJsonFileSync : util.readYamlFileSync;
+    const definitionFile = readFile(filename);
+    const dependencyArns: Map<string, string> = new Map();
+    // Map service name to ARN
+    for (const context of dependenciesDeployContexts) {
+        dependencyArns.set(context.serviceName, context.eventOutputs.lambdaArn);
+    }
+    // Change 'resource' in each state from service name to ARN
+    for (const state_name in definitionFile.States) {
+        const state = definitionFile.States[state_name];
+        if (definitionFile.States.hasOwnProperty(state_name) && 'Resource' in state) {
+            state.Resource = dependencyArns.get(state.Resource);
```

Comment:
Could we add something in the check() method to validate the the lambdas named in the state file actually exist in the Handel file? Right now if you specify a service in your state file that doesn't exist in your Handel file, it will likely fail with a cryptic error for the user.